### ISSUE TITLE
Add M1 support to the unity csharp plugin

### ISSUE THREAD

**Issue by Bart-Verdonck** [Azure-Samples/cognitive-services-speech-sdk#1465]
**Is your feature request related to a problem? Please describe.**
I've tried to build and run the "text-to-speech" quickstart project for unity on an M1 mac, but it can't find/load the dylib in the build.

**Describe the solution you'd like**
Upgrade the sample to use Unity 2021, which has M1 support, and recompile the dylib to also support this architecture

**Describe alternatives you've considered**
I've tried to recompile the macos arm64 x86_64 framework myself to a bundle that can be used in unity but that was unsuccesful.

**Comment by jhakulin** [comment]
@Bart-Verdonck Thanks for the report. It is true that unitypackage is not yet updated to support Mac M1. I created internal workitem to provide the support in future releases. 

In the meantime, you can experiment following
1) Download https://www.nuget.org/packages/Microsoft.CognitiveServices.Speech NuGet. 
2) Unzip nupkg (e.g. with 7zip) and pick dylib from microsoft.cognitiveservices.speech.1.20.0\runtimes\osx-arm64\native directory.
3) Place the dylib under the imported 1.20.0 version of unitypackage under MacOS directory.

Let us know if you have any issue.

**Comment by pankopon** [comment]
Internal work item ref. 4186108.

**Comment by jhakulin** [comment]
@Bart-Verdonck verified the workaround given earlier works on Mac M1, so you can experiment with it. We will target to add this support at May 2022 release.

**Comment by Bart-Verdonck** [comment]
Yeah that dylib works for M1, however when I try to make a build that supports both M1 and Intel, I'm still running in some problems.
Not sure if it's possible to make the two different dylibs into one universal bundle 🤔

**Comment by bh679** [comment]
Do you have macOS support for Unity yet?
